src/config-loader.js

```diff
@@ -0,0 +1,171 @@
+    const result = await dynamicImportDefault(
+      pathToFileURL(configFile).href,
+      fs.statSync(configFile).mtimeMs
+    );
+
+    if (result != null) {
+      return { ...result, configFile };
+    }
+    throw new Error(`invalid export in ${configFile}`);
+  };
+})();
+
+const requireSvelteOptions = (() => {
+  /** @type {NodeRequire} */
+  let esmRequire;
+
+  /**
+   * Try import specified svelte configuration.
+   *
+   * @param {string} configFile
+   * Absolute path of the svelte config file to require.
+   *
+   * @returns {import('@sveltejs/kit').Config}
+   * Returns the svelte configuration object.
+   */
+  return (configFile) => {
+    // identify which require function to use (esm and cjs mode)
+    const requireFn = import.meta.url
+      ? (esmRequire = esmRequire ?? createRequire(import.meta.url))
+      : require;
+
+    // avoid loading cached version on reload
+    delete requireFn.cache[requireFn.resolve(configFile)];
+    const result = requireFn(configFile);
+
+    if (result != null) {
+      return { ...result, configFile };
+    }
+    throw new Error(`invalid export in ${configFile}`);
+  };
+})();
+
+/**
+ * Try find svelte config. First in current working dir otherwise try to
+ * founding it by climbing up the directory tree.
+ *
+ * @returns {string}
+ * Returns an array containing all available config files.
+ */
+function findSvelteConfig() {
+  const lookupDir = process.cwd();
+  let configFiles = getConfigFiles(lookupDir);
+
+  if (configFiles.length === 0) {
+    configFiles = getConfigFilesUp();
+  }
+  if (configFiles.length === 0) {
+    throw new Error('no svelte config found');
+  } else if (configFiles.length > 1) {
+    logger.warn(
+      `found more than one svelte config file, using ${configFiles[0]}. ` +
+        'you should only have one!',
+      configFiles
+    );
+  }
+  return configFiles[0];
+}
+
+/**
+ * Gets the file path of the svelte config by walking up the tree.
+ * Returning the first found. Should solves most of monorepos with
+ * only one config at workspace root.
+ *
+ * @returns {string[]}
+ * Returns an array containing all available config files.
+ */
+function getConfigFilesUp() {
+  const importPath = fileURLToPath(import.meta.url);
+  const pathChunks = path.dirname(importPath).split(path.sep);
+
+  while (pathChunks.length) {
+    pathChunks.pop();
+
+    const parentDir = pathChunks.join(path.posix.sep);
+    const configFiles = getConfigFiles(parentDir);
+
+    if (configFiles.length !== 0) {
+      return configFiles;
+    }
+  }
+  return [];
+}
+
+/**
+ * Gets all svelte config from a specified `lookupDir`.
+ *
+ * @param {string} lookupDir
+ * Directory in which to look for svelte files.
+ *
+ * @returns {string[]}
+ * Returns an array containing all available config files.
+ */
+function getConfigFiles(lookupDir) {
+  return knownConfigFiles
+    .map((candidate) => path.resolve(lookupDir, candidate))
+    .filter((file) => fs.existsSync(file));
```

Comment:
We might gain some perf my making this async, and I don't think it would be a problem since the whole config loading is async anyway. It would make this function a bit more complex, but I think that's worth it.
```suggestion
    .filter((file) => fs.exists(file));
```